### ISSUE TITLE
find_package(tf) mangles catkin_LIBRARIES variable

### ISSUE THREAD

**Issue by roehling** [ros/geometry#51]
The Hydro tf package mangles the catkin_LIBRARIES variable if it is included with `find_package(tf)` instead of `find_package(catkin COMPONENTS tf)`:

```
cmake_minimum_required(VERSION 2.8.3)
project(bug)
find_package(catkin REQUIRED COMPONENTS roscpp)
message(STATUS "catkin_LIBRARIES before: ${catkin_LIBRARIES}")
find_package(tf REQUIRED) # or tf2 
message(STATUS "catkin_LIBRARIES after: ${catkin_LIBRARIES}")
catkin_package()
```

results in

```
-- +++ processing catkin package: 'bug'
-- ==> add_subdirectory(bug)
-- catkin_LIBRARIES before: /opt/ros/hydro/lib/libroscpp.so;/usr/lib/libboost_signals-mt.so;/usr/lib/libboost_filesystem-mt.so;/opt/ros/hydro/lib/librosconsole.so;/opt/ros/hydro/lib/librosconsole_log4cxx.so;/opt/ros/hydro/lib/librosconsole_backend_interface.so;/usr/lib/liblog4cxx.so;/usr/lib/libboost_regex-mt.so;/opt/ros/hydro/lib/libroscpp_serialization.so;/opt/ros/hydro/lib/librostime.so;/usr/lib/libboost_date_time-mt.so;/usr/lib/libboost_system-mt.so;/usr/lib/libboost_thread-mt.so;/usr/lib/x86_64-linux-gnu/libpthread.so;/opt/ros/hydro/lib/libxmlrpcpp.so;/opt/ros/hydro/lib/libcpp_common.so;/opt/ros/hydro/lib/libconsole_bridge.so
-- catkin_LIBRARIES after: 
-- Configuring done
```

I have no idea where this comes from, but at least I can tell you that it works with `tf2_bullet` and `geometry_msgs`, so it's not a general catkin issue.

**Comment by dirk-thomas** [comment]
This is not `tf`s fault but `actionlib_msgs`: https://github.com/ros/common_msgs/blob/hydro-devel/actionlib_msgs/cmake/actionlib_msgs-extras.cmake.em#L2

**Comment by dirk-thomas** [comment]
Closed in favor of ros/common_msgs#27.
